python/tests/unit/agents/test_chat_completion_agent.py

```diff
@@ -0,0 +1,179 @@
+
+    messages = [message async for message in agent.invoke(history)]
+
+    assert len(messages) == 1
+    assert messages[0].content == "Processed Message"
+
+
+@pytest.mark.asyncio
+async def test_invoke_tool_call_added():
+    kernel = create_autospec(Kernel)
+    chat_completion_service = create_autospec(ChatCompletionClientBase)
+    kernel.get_service.return_value = chat_completion_service
+    agent = ChatCompletionAgent(kernel=kernel, service_id="test_service", name="Test Agent")
+
+    history = ChatHistory(messages=[ChatMessageContent(role=AuthorRole.USER, content="Initial Message")])
+
+    async def mock_get_chat_message_contents(chat_history, settings, kernel, arguments):
+        new_messages = [
+            ChatMessageContent(role=AuthorRole.SYSTEM, content="Processed Message 1"),
+            ChatMessageContent(role=AuthorRole.TOOL, content="Processed Message 2"),
+        ]
+        chat_history.messages.extend(new_messages)
+        return new_messages
+
+    chat_completion_service.get_chat_message_contents = AsyncMock(side_effect=mock_get_chat_message_contents)
+
+    messages = [message async for message in agent.invoke(history)]
+
+    assert len(messages) == 2
+    assert messages[0].content == "Processed Message 1"
+    assert messages[1].content == "Processed Message 2"
+
+    assert len(history.messages) == 3
+    assert history.messages[1].content == "Processed Message 1"
+    assert history.messages[2].content == "Processed Message 2"
+    assert history.messages[1].name == "Test Agent"
+    assert history.messages[2].name == "Test Agent"
+
+
+@pytest.mark.asyncio
+async def test_invoke_no_service_throws():
+    kernel = create_autospec(Kernel)
+    kernel.get_service.return_value = None
+    agent = ChatCompletionAgent(kernel=kernel, service_id="test_service", name="Test Agent")
+
+    history = ChatHistory(messages=[ChatMessageContent(role=AuthorRole.USER, content="Initial Message")])
+
+    with pytest.raises(KernelServiceNotFoundError):
+        async for _ in agent.invoke(history):
+            pass
+
+
+@pytest.mark.asyncio
+async def test_invoke_stream():
+    kernel = create_autospec(Kernel)
+    kernel.get_service.return_value = create_autospec(ChatCompletionClientBase)
+
+    agent = ChatCompletionAgent(kernel=kernel, service_id="test_service", name="Test Agent")
+
+    history = ChatHistory(messages=[ChatMessageContent(role=AuthorRole.USER, content="Initial Message")])
+
+    with patch(
+        "semantic_kernel.connectors.ai.chat_completion_client_base.ChatCompletionClientBase.get_streaming_chat_message_contents",
+        return_value=AsyncMock(),
+    ) as mock:
+        mock.return_value.__aiter__.return_value = [
+            [ChatMessageContent(role=AuthorRole.USER, content="Initial Message")]
+        ]
+
+        async for message in agent.invoke_stream(history):
+            assert message.role == AuthorRole.USER
+            assert message.content == "Initial Message"
```

Comment:
would be good to also assert some things about the history after running, for all the tests...